src/compiler.js

```diff
@@ -0,0 +1,36 @@
+import { resolve } from 'path';
+import promisify from 'es6-promisify';
+import rimraf from 'rimraf';
+import prerender from './lib/prerender';
+import webpackConfig from './lib/webpack-config';
+import runWebpack, { showStats } from './lib/run-webpack';
+
+class Compiler {
```

Comment:
(Can anyone make an opinion?)
The implementation looks awesome, but as a personal choice, I'd prefer non-class approaches 👍  
I guess it depends on the project... are classes used in `preact-cli`?
If so, it could be good to keep consistency with pojo/factory approach 😃